joy_teleop/joy_teleop/joy_teleop.py

```diff
@@ -47,6 +46,278 @@ class JoyTeleopException(Exception):
+    ml = member.split('-')
+    if len(ml) < 1:
+        return
+    target = msg
+    for i in ml[:-1]:
+        target = getattr(target, i)
+    setattr(target, ml[-1], value)
+
+
+class JoyTeleopCommand:
+
+    def __init__(self, name, config, button_name, axes_name):
+        self.buttons = []
+        if button_name in config:
+            self.buttons = config[button_name]
+        self.axes = []
+        if axes_name in config:
+            self.axes = config[axes_name]
+
+        if len(self.buttons) == 0 and len(self.axes) == 0:
+            raise JoyTeleopException("No buttons or axes configured for command '{}'".format(name))
+
+        # This is used to short-circuit the run command if there aren't enough
+        # buttons in the message
+        self.min_button = None
+        if len(self.buttons) > 0:
+            self.min_button = min(self.buttons)
+        self.min_axis = None
+        if len(self.axes) > 0:
+            self.min_axis = min(self.axes)
+
+        # This is used to "debounce" the message; if we get multiple presses
+        # of the button(s), we only activate on the first one until it toggles
+        # again.  If there are multiple buttons, this means that mashing all of
+        # them together will only activate this topic once.
+        self.active = 0
+
+        # We use 'ready' to deal with cases where a service/action becomes
+        # ready between the last call and the current one.  That is, if we get
+        # a joystick command to call the service/action while it is not ready,
+        # followed immediately by another command to call the service/action
+        # and it has become ready, we still want to call it the second time.
+        self.ready = False
+
+    def check_buttons_and_axes(self, joy_state):
+        if (self.min_button is not None and len(joy_state.buttons) <= self.min_button) and \
+           (self.min_axis is not None and len(joy_state.axes) <= self.min_axis):
+            # There weren't enough buttons or axes in the message, so it can't possibly
+            # be a message for us.
+            return False
+
+        last_active = self.active
+
+        self.active = 0
+
+        for button in self.buttons:
+            try:
+                self.active |= joy_state.buttons[button] == 1
+            except IndexError:
+                # We can get an index error if we are configured for multiple
+                # buttons like (0, 10), but the length of the joystick buttons
+                # is only 1.  Just ignore these.
+                pass
+
+        for axis in self.axes:
+            try:
+                self.active |= joy_state.axes[axis] == 1.0
+            except IndexError:
+                # We can get an index error if we are configured for multiple
+                # buttons like (0, 10), but the length of the joystick buttons
+                # is only 1.  Just ignore these.
+                pass
+
+        if self.active == 0:
```

Comment:
@clalancette should this be
```suggestion
        if not self.active:
```
?

Reply:
Fixed in the latest.